### PR TITLE
Updated application/entity generation

### PR DESCRIPTION
**Pull request by MathieuAA**
- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [X] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [X] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

Used provided ways to get the config file (`.yo-rc.json`) content to be read and used by the generators.

### CODE REVIEW

File: generators/generator-base-private.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ const path = require('path');
 const _ = require('lodash');
 const fs = require('fs');
 const Generator = require('yeoman-generator');
+const Storage = require('yeoman-generator/lib/util/storage');
```
Comment (deepu105) [inline generators/generator-base-private.js]: @MathieuAA is this public API or internal API?

File: generators/generator-base-private.js [hunk 2]
```diff
@@ -54,6 +55,7 @@ module.exports = class extends Generator {
         this.env.options.appPath = this.config.get('appPath') || CLIENT_MAIN_SRC_DIR;
         // expose lodash to templates
         this._ = _;
+        this.createConfigFromNewConfFile();
```
Comment (deepu105) [inline generators/generator-base-private.js]: @MathieuAA why is this needed, could you please explain?

File: generators/generator-base-private.js (hunk 1 above)
Comment (deepu105) [reply generators/generator-base-private.js]: ok

File: generators/generator-base-private.js (hunk 2 above)
Comment (deepu105) [reply generators/generator-base-private.js]: you need this only for import-jdl right? i'm worried about the blueprints, as I dont know what side effect this would add to all generators

Comment (MathieuAA) [reply generators/generator-base-private.js]: I think I caught your meaning, I can try to make it only possible for the import-jdl subgen, now that I understand how the Storage works.

**Comment by JulioJu** [comment]
@MathieuAA thanks, fix, perfectly https://github.com/jhipster/generator-jhipster/issues/7775 !

**Comment by JulioJu** [comment]
Sorry, this commit introduce a new bug. See https://github.com/jhipster/generator-jhipster/issues/7775#issuecomment-397980893.
